arraycontext/impl/pytato/compile.py

```diff
@@ -262,6 +262,8 @@ class BaseLazilyCompilingFunctionCaller:
     program_cache: Dict["PMap[Tuple[Any, ...], AbstractInputDescriptor]",
                         "CompiledFunction"] = field(default_factory=lambda: {})
 
+    single_version_only: bool = False
```

Comment:
I'm worried this might lead to developer errors. I would prefer if we don't default and ensure that the current version is passed during instantiation.

Reply:
I removed the default in 5ed25a5

Reply:
This fails in grudge now:

```
  File "wave-mpi-lazy.py", line 201, in main
    compiled_rhs = actx.compile(rhs)
                   ^^^^^^^^^^^^^^^^^
  File "/home/runner/work/arraycontext/arraycontext/mirgecom/src/grudge/grudge/array_context.py", line 416, in compile
    return _DistributedLazilyPyOpenCLCompilingFunctionCaller(self, f)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: BaseLazilyCompilingFunctionCaller.__init__() missing 1 required positional argument: 'single_version_only'
```

Reply:
Hmm... Maybe then we introduce the default as you suggested, but have a deprecation period attached to it?